firefox-ios/Client/Frontend/Settings/AppearanceSettings/AppearanceSettingsView.swift

```diff
@@ -42,44 +42,35 @@ struct AppearanceSettingsView: View, FeatureFlaggable {
 
     private struct UX {
         static let spacing: CGFloat = 24
+        static let cornerRadius: CGFloat = 24
+        static var spacingCurrentOS: CGFloat {
+            guard #available(iOS 26.0, *) else { return 0 }
+            return UX.spacing
+        }
     }
 
     var body: some View {
         ScrollView {
-            VStack {
+            VStack(spacing: UX.spacingCurrentOS) {
                 // Section for selecting the browser theme.
-                GenericSectionView(theme: currentTheme,
-                                   title: String.BrowserThemeSectionHeader,
-                                   identifier: AccessibilityIdentifiers.Settings.Appearance.browserThemeSectionTitle) {
-                    ThemeSelectionView(theme: currentTheme,
-                                       selectedThemeOption: themeOption,
-                                       onThemeSelected: updateBrowserTheme)
-                }
+                BrowserThemeSection(
+                    theme: currentTheme,
+                    themeOption: themeOption,
+                    onThemeSelected: updateBrowserTheme,
+                    cornerRadius: UX.cornerRadius
+                )
                 // Section for toggling website appearance (e.g., dark mode).
-                GenericSectionView(theme: currentTheme,
-                                   title: String.WebsiteAppearanceSectionHeader,
-                                   description: String.WebsiteDarkModeDescription,
-                                   identifier: AccessibilityIdentifiers.Settings.Appearance.websiteAppearanceSectionTitle) {
-                    DarkModeToggleView(theme: currentTheme,
-                                       isEnabled: NightModeHelper.isActivated(),
-                                       onChange: setWebsiteDarkMode)
-                }
+                WebsiteAppearanceSection(theme: currentTheme, onChange: setWebsiteDarkMode, cornerRadius: UX.cornerRadius)
+
                 if shouldShowPageZoom {
-                    GenericSectionView(theme: currentTheme,
-                                       title: .Settings.Appearance.PageZoom.SectionHeader,
-                                       identifier: .Settings.Appearance.PageZoom.SectionHeader) {
-                        GenericItemCellView(title: .Settings.Appearance.PageZoom.PageZoomTitle,
-                                            image: .chevronRightLarge,
-                                            theme: currentTheme) {
-                            delegate?.pressedPageZoom()
-                        }
+                    PageZoomSection(theme: currentTheme, cornerRadius: UX.cornerRadius) {
+                        delegate?.pressedPageZoom()
                     }
                 }
                 Spacer()
             }
         }
-        .padding(.top, UX.spacing)
-        .frame(maxWidth: .infinity)
+        .modifier(PaddingStyle(theme: currentTheme, spacing: UX.spacing))
```

Comment:
this differs a bit from @dicarobinho PR, but I didn't see any issues.

Ref: 
<img width="583" height="118" alt="Screenshot 2025-08-15 at 11 46 56 AM" src="https://github.com/user-attachments/assets/355b9162-fde6-4cb6-ab09-cd59ff7ce311" />